lib/dry/schema/macros/array.rb

```diff
@@ -10,10 +10,10 @@ module Macros
       # @api private
       class Array < DSL
         # @api private
-        def value(*args, **opts, &block)
+        def value(type_spec, *args, **opts, &block)
           type(:array)
 
-          extract_type_spec(*args, set_type: false) do |*predicates, type_spec:, type_rule:|
+          extract_type_spec(type_spec, *args, set_type: false) do |*predicates, type_spec:, type_rule:|
             type(schema_dsl.array[type_spec]) if type_spec
 
             is_hash_block = type_spec.equal?(:hash)
```

Comment:
This is quite tricky to handle, since this isn't always going to be true—it's possible to have a hash type_spec that takes no block.